massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -206,6 +314,42 @@ impl SpeculativeLedger {
             )));
         }
 
+        if bytecode.len() > self.max_bytecode_size as usize {
+            return Err(ExecutionError::RuntimeError(format!(
+                "could not set bytecode for address {}: bytecode size exceeds maximum allowed size",
+                addr
+            )));
+        }
+
+        if let Some(old_bytecode_size) = self.get_bytecode(addr).map(|b| b.len()) {
+            let diff_size_storage: i64 = (bytecode.len() as i64) - (old_bytecode_size as i64);
+            let storage_cost_bytecode = self
+                .storage_costs_constants
+                .ledger_cost_per_byte
+                .checked_mul_u64(diff_size_storage.unsigned_abs())
+                .ok_or_else(|| {
+                    ExecutionError::RuntimeError("try to store too much data".to_string())
+                })?;
+
+            match diff_size_storage.cmp(&0) {
+                Ordering::Greater => {
+                    self.transfer_coins(Some(*addr), None, storage_cost_bytecode)?
+                }
+                Ordering::Less => self.transfer_coins(None, Some(*addr), storage_cost_bytecode)?,
+                Ordering::Equal => {}
+            };
+        } else {
```

Comment:
I think the "else" case should never happen because "!self.entry_exists(addr)" was tested above

Reply:
The address can exist but have no bytecode for now.

Reply:
in that case it always has a bytecode but of length 0 I think

Reply:
Just checked and that the case but if we change this behavior in the future and forget to update this part of the code we could lead to free bytecode. I don't think it's a problem to let here to be robust.

Reply:
ok